Aplib.Core/Desire/DesireSets/DesireSet.cs

```diff
@@ -93,5 +108,20 @@ public void Update(TBeliefSet beliefSet)
                 _goalStructureStack.Pop();
             }
         }
+
+        /// <summary>
+        /// Implicitly lifts a goal into a desire set.
+        /// </summary>
+        /// <inheritdoc cref="LiftingExtensionMethods.Lift{TBeliefSet}(IGoal{TBeliefSet})" path="/param[@name='goal']"/>
+        /// <returns>The most logically matching desire set, wrapping around <paramref name="goal"/>.</returns>
+        public static implicit operator DesireSet<TBeliefSet>(Goal<TBeliefSet> goal) => goal.Lift();
```

Comment:
This is now implicitly converted, since goal.Lift() returns a goal structure. Could you do this explicitly for clarity?
```suggestion
        public static implicit operator DesireSet<TBeliefSet>(Goal<TBeliefSet> goal) => goal.Lift().Lift();
```

Reply:
So I should not use the feature I am implementing?

Reply:
My point is not about using features or not. In my opinion, calling Lift two times makes it immediately clear what the code is doing, while using one explicit lift and one implicit lift is not immediately clear I think. The first thing I checked when I looked at this code is if there was a Lift method lifting a Goal to a DesireSet (this is not even possible in C#, but I still looked for it). It just makes the code a little bit more clear IMO. Bonus: it removes a function call.